style/authentication/README.md

```diff
@@ -0,0 +1,27 @@
+# Authentication
+
+When building applications that use authentication, security is crucial. To help ensure that we are building secure applications, we follow the [OWASP Authentication General Guidelines](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Authentication_General_Guidelines).
+
+Because security is a constantly moving target, we recommend that you [don't roll your own authentication](https://blog.codeship.com/why-you-shouldnt-roll-your-own-authentication/), and instead rely on a battle-tested authentication mechanism. As an example, for Rails apps we use [Devise](https://github.com/plataformatec/devise).
+
+## Password Requirements
+
+We follow the [OWASP Password Guidelines](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Implement_Proper_Password_Strength_Controls) for strong passwords:
+
+> The password change mechanism should require a minimum level of complexity that makes sense for the application and its user population. For example:
+>
+> Password must meet at least 3 out of the following 4 complexity rules
+>
+> - at least 1 uppercase character (A-Z)
+> - at least 1 lowercase character (a-z)
+> - at least 1 digit (0-9)
+> - at least 1 special character (punctuation) — do not forget to treat space as special characters too
```

Comment:
I copied this section from the OWASP guidelines, but I agree that the wording could be improved. I'll update it.